pkg/controller/migmigration/task.go

```diff
@@ -1031,20 +1031,22 @@ func (t *Task) init() error {
 
 	t.initPipeline()
 
-	hasImageStreams, err := t.hasImageStreams()
-	if err != nil {
-		return err
-	}
+	if t.stage() {
+		hasImageStreams, err := t.hasImageStreams()
+		if err != nil {
+			return err
+		}
 
-	anyPVs, _ := t.hasPVs()
-	if t.stage() && (!anyPVs && !hasImageStreams) {
-		t.Owner.Status.SetCondition(migapi.Condition{
-			Type:     StageNoOp,
-			Status:   True,
-			Category: migapi.Warn,
-			Message:  "Stage migration was run without any PVs or ImageStreams in source cluster. No Velero operations were initiated.",
-			Durable:  true,
-		})
+		anyPVs, _ := t.hasPVs()
```

Comment:
@shawn-hurley do you think it would be feasible in this PR to add one additional check?

```
if !t.Owner.Status.HasCondition(StageNoOp)
   // only run hasPVs() and hasImageStreams() if this condition hasn't already been checked
```

Reply:
yes I do give me a second

Reply:
Added let me know if it looks right

Reply:
LGTM